src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildADesktopLibrary.cs

```diff
@@ -459,5 +459,39 @@ public void It_marks_extension_references_as_externally_resolved(bool? markAsExt
             frameworkReferenceCount.Should().BeGreaterThan(0);
             extensionReferenceCount.Should().BeGreaterThan(0);
         }
+
+        [FullMSBuildOnlyFact]
+        public void It_tolerates_newline_in_hint_path()
+        {
+            TestProject project = new TestProject()
+            {
+                Name = "NETFrameworkLibrary",
+                TargetFrameworks = "net462",
+                IsSdkProject = true
+            };
+
+            var testAsset = _testAssetsManager.CreateTestProject(project, "SimpleNamesWithHintPathsWithNewLines")
+                .WithProjectChanges((path, p) =>
+                {
+                    if (Path.GetFileNameWithoutExtension(path) == project.Name)
+                    {
+                        var ns = p.Root.Name.Namespace;
+
+                        var itemGroup = new XElement(ns + "ItemGroup");
+                        p.Root.Add(itemGroup);
+
+                        itemGroup.Add(
+                            new XElement(ns + "Reference",
+                                new XAttribute("Include", "System.Net.Http"),
+                                new XElement("HintPath", $"{Environment.NewLine}net461\\lib\\System.Net.Http.dll{Environment.NewLine}")));
```

Comment:
In this test, the HintPath won't be used (as there's no DLL at that path).  Instead the DLL will resolve to the one in the targeting pack.

To more closely match the user scenario here, you could compile one project, get the full path to its DLL in the output path, and use that HintPath in a Reference in a second project.

Reply:
Create a project as suggested